rinja_parser/src/lib.rs

```diff
@@ -727,14 +727,23 @@ impl Level {
     const MAX_DEPTH: u8 = 128;
 }
 
-#[allow(clippy::type_complexity)]
 fn filter(i: &str, level: Level) -> ParseResult<'_, (&str, Option<Vec<WithSpan<'_, Expr<'_>>>>)> {
-    let (i, (_, fname, args)) = tuple((
-        char('|'),
-        ws(identifier),
-        opt(|i| Expr::arguments(i, level, false)),
-    ))(i)?;
-    Ok((i, (fname, args)))
+    let (j, _) = take_till(not_ws)(i)?;
+    let had_spaces = i.len() != j.len();
+    let (j, _) = pair(char('|'), not(char('|')))(j)?;
+
+    if !had_spaces {
+        cut(pair(
+            ws(identifier),
+            opt(|i| Expr::arguments(i, level, false)),
+        ))(j)
+    } else {
+        Err(nom::Err::Failure(ErrorContext::new(
+            "The filter operator `|` must not be preceeded by any whitespace characters.\n\
```

Comment:
Little nit: other error messages don't start with a capital letter.

Reply:
I was not sure about that, because it's two sentences. Maybe I should join them with a ";", so an initial lower cases character makes more sense? :D